test/e2e/custom_node_selector_tests.go

```diff
@@ -0,0 +1,80 @@
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/kube-reporting/metering-operator/test/reportingframework"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func testNodeSelectorConfigurationWorks(t *testing.T, rf *reportingframework.ReportingFramework) {
```

Comment:
Open to suggestions on how to make this test a bit more robust. At the moment, we just verify that the node selector configuration that was specified in the MeteringConfig custom resource has propagated to the metering operand Pods. We could do this kind of test, and then patch that same MeteringConfig custom resource with the intention of removing those node selector configurations, waiting for those changes to reconcile, and validating those node selector configurations no longer exist.

Reply:
This seems fine for now. It would have caught the case where the move to v1 was wiping settings.